ServiceTier/Services/TaskService.cs

```diff
@@ -82,15 +83,19 @@ public void UpdateTask(TaskDTO taskDTO)
             unitOfWork.Save();
         }
 
-        public void UpdateTaskStatus(int taskId, bool newStatus)
+        public void UpdateTaskStatus(TaskDTO model)
         {
-            Tasks task = unitOfWork.Tasks.GetByID(taskId);
-            if (task == null)
+            if (model.IsRunning)
```

Comment:
so basically model.IsRunning = !model.IsRunning? why are we flipping this value?

Reply:
Yes, I will change to model.IsActive = !model.IsActive; Also, IsRunning was a mistake by me. I meant to flip the status of task to archive it and make active again.

Reply:
still not sure why we are just flipping that flag on every update. can you explain?

Reply:
Well I had envisioned that on our tasks page only active tasks will be listed. Each task will have the button to archive that task . That button will trigger this UpdateStatus() method and IsActive property of underlying task will be changed to false. Then the task will be displayed only on our archive page.

Reply:
And vice versa when making archived task active again.